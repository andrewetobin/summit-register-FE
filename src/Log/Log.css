.Log {
  background: url("../utilities/Images/mountains-2.jpg") no-repeat center
  rgba(165, 165, 165, 0.68);
  background-size: auto 100%;
  overflow: scroll;
}

h4 {
  font-size: 14px;
}

.comment-container {
  margin: 0 5%;
  width: 90%;
}

.comment {
  background: grey;
  border: 1px solid white;
  border-radius: 20px;
  margin: 10px;
  opacity: .9;
  padding: 10px;
  text-align: left;
}

.empty-log-msg {
  border: 1px solid white;
  border-radius: 20px;
  background: grey;
  height: 10%;
  margin: 25vh auto 0 auto;
  opacity: .9;
  padding: 20px;
  text-align: center;
  width: 50%;
}
