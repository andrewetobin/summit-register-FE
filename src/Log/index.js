import React, { Component } from 'react';
import './Log.css';

class Log extends Component {
  render() {
    return (
      <div className="Log">
      </div>
    );
  }
}

export default Log;