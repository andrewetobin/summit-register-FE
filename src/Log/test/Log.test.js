import React from "react";
import ReactDOM from "react-dom";
import Log from "..";
import { shallow } from "enzyme";

describe("Log", () => {
  let wrapper;
  const currentMountainLog = [
    {
      id: "1",
      type: "registry",
      attributes: {
        name: "Andrew Mockett",
        hometown: "Boulder",
        comments: "Great hike",
        sign_time: "2018-12-19 21:10:17 UTC"
      }
    },
    {
      id: "2",
      type: "registry",
      attributes: {
        name: "Andrew Tobin",
        hometown: "Denver",
        comments: "Best hike ever",
        sign_time: "2018-12-19 21:11:29 UTC"
      }
    }
  ];
  beforeEach(() => {
    wrapper = shallow(<Log currentMountainLog={currentMountainLog} />);
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Log currentMountainLog={currentMountainLog} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  describe("empty log message", () => {
    it("should not display empty log msg when entries present", () => {
      expect(wrapper.find(".empty-log-msg")).toHaveLength(0);
    });

    it("should display empty log msg when zero entries", () => {
      wrapper = shallow(<Log currentMountainLog={[]} />);
      expect(wrapper.find(".empty-log-msg")).toHaveLength(1);
    });
  });
});
