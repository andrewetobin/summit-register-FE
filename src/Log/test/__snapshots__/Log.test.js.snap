// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Log should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Log
    currentMountainLog={
      Array [
        Object {
          "attributes": Object {
            "comments": "Great hike",
            "hometown": "Boulder",
            "name": "Andrew Mockett",
            "sign_time": "2018-12-19 21:10:17 UTC",
          },
          "id": "1",
          "type": "registry",
        },
        Object {
          "attributes": Object {
            "comments": "Best hike ever",
            "hometown": "Denver",
            "name": "Andrew Tobin",
            "sign_time": "2018-12-19 21:11:29 UTC",
          },
          "id": "2",
          "type": "registry",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1 />,
        <section
          className="comment-container"
        >
          <div
            className="comment"
          >
            <h4>
              Name: 
              Andrew Tobin
            </h4>
            <h4>
              Hometown: 
              Denver
            </h4>
            <h4>
              Comments: 
              Best hike ever
               
            </h4>
            <h4>
              Date: 
              2018-12-19 21:11:29 UTC
            </h4>
          </div>
          <div
            className="comment"
          >
            <h4>
              Name: 
              Andrew Mockett
            </h4>
            <h4>
              Hometown: 
              Boulder
            </h4>
            <h4>
              Comments: 
              Great hike
               
            </h4>
            <h4>
              Date: 
              2018-12-19 21:10:17 UTC
            </h4>
          </div>
        </section>,
      ],
      "className": "Log",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="comment"
            >
              <h4>
                Name: 
                Andrew Tobin
              </h4>
              <h4>
                Hometown: 
                Denver
              </h4>
              <h4>
                Comments: 
                Best hike ever
                 
              </h4>
              <h4>
                Date: 
                2018-12-19 21:11:29 UTC
              </h4>
            </div>,
            <div
              className="comment"
            >
              <h4>
                Name: 
                Andrew Mockett
              </h4>
              <h4>
                Hometown: 
                Boulder
              </h4>
              <h4>
                Comments: 
                Great hike
                 
              </h4>
              <h4>
                Date: 
                2018-12-19 21:10:17 UTC
              </h4>
            </div>,
          ],
          "className": "comment-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h4>
                  Name: 
                  Andrew Tobin
                </h4>,
                <h4>
                  Hometown: 
                  Denver
                </h4>,
                <h4>
                  Comments: 
                  Best hike ever
                   
                </h4>,
                <h4>
                  Date: 
                  2018-12-19 21:11:29 UTC
                </h4>,
                undefined,
              ],
              "className": "comment",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Name: ",
                    "Andrew Tobin",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Name: ",
                  "Andrew Tobin",
                ],
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Hometown: ",
                    "Denver",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Hometown: ",
                  "Denver",
                ],
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Comments: ",
                    "Best hike ever",
                    " ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Comments: ",
                  "Best hike ever",
                  " ",
                ],
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Date: ",
                    "2018-12-19 21:11:29 UTC",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Date: ",
                  "2018-12-19 21:11:29 UTC",
                ],
                "type": "h4",
              },
              undefined,
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h4>
                  Name: 
                  Andrew Mockett
                </h4>,
                <h4>
                  Hometown: 
                  Boulder
                </h4>,
                <h4>
                  Comments: 
                  Great hike
                   
                </h4>,
                <h4>
                  Date: 
                  2018-12-19 21:10:17 UTC
                </h4>,
                undefined,
              ],
              "className": "comment",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Name: ",
                    "Andrew Mockett",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Name: ",
                  "Andrew Mockett",
                ],
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Hometown: ",
                    "Boulder",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Hometown: ",
                  "Boulder",
                ],
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Comments: ",
                    "Great hike",
                    " ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Comments: ",
                  "Great hike",
                  " ",
                ],
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Date: ",
                    "2018-12-19 21:10:17 UTC",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Date: ",
                  "2018-12-19 21:10:17 UTC",
                ],
                "type": "h4",
              },
              undefined,
            ],
            "type": "div",
          },
        ],
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1 />,
          <section
            className="comment-container"
          >
            <div
              className="comment"
            >
              <h4>
                Name: 
                Andrew Tobin
              </h4>
              <h4>
                Hometown: 
                Denver
              </h4>
              <h4>
                Comments: 
                Best hike ever
                 
              </h4>
              <h4>
                Date: 
                2018-12-19 21:11:29 UTC
              </h4>
            </div>
            <div
              className="comment"
            >
              <h4>
                Name: 
                Andrew Mockett
              </h4>
              <h4>
                Hometown: 
                Boulder
              </h4>
              <h4>
                Comments: 
                Great hike
                 
              </h4>
              <h4>
                Date: 
                2018-12-19 21:10:17 UTC
              </h4>
            </div>
          </section>,
        ],
        "className": "Log",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="comment"
              >
                <h4>
                  Name: 
                  Andrew Tobin
                </h4>
                <h4>
                  Hometown: 
                  Denver
                </h4>
                <h4>
                  Comments: 
                  Best hike ever
                   
                </h4>
                <h4>
                  Date: 
                  2018-12-19 21:11:29 UTC
                </h4>
              </div>,
              <div
                className="comment"
              >
                <h4>
                  Name: 
                  Andrew Mockett
                </h4>
                <h4>
                  Hometown: 
                  Boulder
                </h4>
                <h4>
                  Comments: 
                  Great hike
                   
                </h4>
                <h4>
                  Date: 
                  2018-12-19 21:10:17 UTC
                </h4>
              </div>,
            ],
            "className": "comment-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h4>
                    Name: 
                    Andrew Tobin
                  </h4>,
                  <h4>
                    Hometown: 
                    Denver
                  </h4>,
                  <h4>
                    Comments: 
                    Best hike ever
                     
                  </h4>,
                  <h4>
                    Date: 
                    2018-12-19 21:11:29 UTC
                  </h4>,
                  undefined,
                ],
                "className": "comment",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Name: ",
                      "Andrew Tobin",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Name: ",
                    "Andrew Tobin",
                  ],
                  "type": "h4",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Hometown: ",
                      "Denver",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Hometown: ",
                    "Denver",
                  ],
                  "type": "h4",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Comments: ",
                      "Best hike ever",
                      " ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Comments: ",
                    "Best hike ever",
                    " ",
                  ],
                  "type": "h4",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Date: ",
                      "2018-12-19 21:11:29 UTC",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Date: ",
                    "2018-12-19 21:11:29 UTC",
                  ],
                  "type": "h4",
                },
                undefined,
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h4>
                    Name: 
                    Andrew Mockett
                  </h4>,
                  <h4>
                    Hometown: 
                    Boulder
                  </h4>,
                  <h4>
                    Comments: 
                    Great hike
                     
                  </h4>,
                  <h4>
                    Date: 
                    2018-12-19 21:10:17 UTC
                  </h4>,
                  undefined,
                ],
                "className": "comment",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Name: ",
                      "Andrew Mockett",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Name: ",
                    "Andrew Mockett",
                  ],
                  "type": "h4",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Hometown: ",
                      "Boulder",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Hometown: ",
                    "Boulder",
                  ],
                  "type": "h4",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Comments: ",
                      "Great hike",
                      " ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Comments: ",
                    "Great hike",
                    " ",
                  ],
                  "type": "h4",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Date: ",
                      "2018-12-19 21:10:17 UTC",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Date: ",
                    "2018-12-19 21:10:17 UTC",
                  ],
                  "type": "h4",
                },
                undefined,
              ],
              "type": "div",
            },
          ],
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
